### PR TITLE
ZK-3624: Spinner - improve initial value

### PR DESCRIPTION
**Pull request by charlesqqq**

### CODE REVIEW

File: zktest/src/archive/test2/F85-ZK-3624.zul [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	<label multiline="true">
+		1. Click spinner up button, the value should be 1
+		2. Empty spinner text
+		3. Click button setConstraint("min 0")
+		4. Click spinner up button, the value should be 1
+		5. Empty spinner text
+		6. Click button setConstraint("min 2")
+		7. Click spinner up button, the value should be 2
+	</label>
+	<spinner id="sp" constraint="min -1"/>
```
Comment (scribetw) [inline zktest/src/archive/test2/F85-ZK-3624.zul]: consider max condition ex: `max -10` ?

File: zktest/test/java/org/zkoss/zktest/zats/test2/F85_ZK_3624Test.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		connect();
+		
+		increase();
+		Assert.assertEquals("1", jq(".z-spinner-input").val());
+		clearSpinnerValue();
+		
+		click(jq("@button:eq(0)"));
+		waitResponse();
+		
+		increase();
```
Comment (scribetw) [inline zktest/test/java/org/zkoss/zktest/zats/test2/F85_ZK_3624Test.java]: Test decrease situation too, not only increase

File: zul/src/archive/web/js/zul/inp/Spinner.js [hunk 3]
```diff
@@ -108,6 +108,7 @@ zul.inp.Spinner = zk.$extends(zul.inp.NumberInputWidget, {
 		if (!this._buttonVisible || this._disabled) return;
 
 		var btn = this.$n('btn');
+		this._noPreviousValue = this.getInputNode().value == '';
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/inp/Spinner.js]: What if we trigger increase/decrease by keyboard?

**Comment by scribetw** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io/reviews/zkoss/zk/1122)*

**Comment by wenninghsu** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I could not rebase F85-ZK-3624 with master. There are conflicts.

**Pull request merged**